### DISCUSSION TITLE
Having more calendar fields accessible through context

### DISCUSSION THREAD

**Discussion by ImXico** [gpbl/react-day-picker#2562] [Support] [answered]
Hey guys! First off thanks for the great work on this library.

I was wondering — is there a reason why fields such as `showOutsideDays` and `numberOfWeeks` are not part of the context that is accessible via `useDayPicker`? 

Context: I'm using custom components for both `Nav` and `DayButton` — having the aforementioned fields available through context would make it so these custom components wouldn't need to be mounted and unmounted constantly. 

E.g.:

```tsx
// ideally `MyCustomButton` would be able to access config fields such as `showOutsideDays` via context
DayButton: MyCustomDayButton

// right now that is not the case, so have to do this, which:
// - causes many mounts/unmounts, and
// - breaks custom select/deselect button styles within my custom button component (because of those mounts/unmounts)
DayButton: props => <MyCustomDayButton {...props} showOutsideDays={showOutsideDays} />
```

**Comment by gpbl** [comment]
@ImXico that's a good suggestions, we can make it return the props.

**Comment by gpbl** [comment, marked as answer]
I’ve added a `dayPickerProps` to the values returned by `useDayPicker` in https://github.com/gpbl/react-day-picker/pull/2572.

Would it work for you, @ImXico?

**Reply by ImXico** [reply]
@gpbl yeah that's perfect, thank you!
